Modules/Registration/Metricsv4/include/itkJointHistogramMutualInformationImageToImageMetricv4.hxx

```diff
@@ -90,54 +90,62 @@ JointHistogramMutualInformationImageToImageMetricv4<TFixedImage,
    *  max and min for the fixed image. */
   itk::ImageRegionConstIteratorWithIndex<TFixedImage> fi(this->m_FixedImage, this->m_FixedImage->GetRequestedRegion());
 
-  /** \todo multi-thread me */
-  while (!fi.IsAtEnd())
   {
-    typename TFixedImage::PointType fixedSpacePhysicalPoint;
-    this->m_FixedImage->TransformIndexToPhysicalPoint(fi.GetIndex(), fixedSpacePhysicalPoint);
-    /* A null mask implies entire space is to be used.*/
-    if (this->m_FixedImageMask.IsNull() || this->m_FixedImageMask->IsInsideInWorldSpace(fixedSpacePhysicalPoint))
+    const typename Superclass::FixedImageMaskType::WorldSpaceContext worldSpaceContext(Superclass::m_FixedImageMask);
+
+    /** \todo multi-thread me */
+    while (!fi.IsAtEnd())
     {
```

Comment:
Should [MinimumMaximumImageCalculator](https://itk.org/Doxygen/html/classitk_1_1MinimumMaximumImageCalculator.html) be used here? And parallelize it? Though parallelization might not matter much in this cases, as this is almost certainly limited by memory throughput, not computation.

Reply:
I recall optimizing the itk::MinimumMaximumImageFilter and was able to get surprising gains in the filter.

Reply:
As far as I can see, MinimumMaximumImageCalculator does not support masks: https://github.com/InsightSoftwareConsortium/ITK/blob/8010fafde3eb40a69db6ee9f49e547d0565da92b/Modules/Core/Common/include/itkMinimumMaximumImageCalculator.h

Reply:
Ah, I didn't think about that 😄